dmoj/executors/compiled_executor.py

```diff
@@ -95,6 +98,14 @@ def create_files(self, problem_id: str, source_code: bytes, *args, **kwargs) ->
         with open(self._code, 'wb') as fo:
             fo.write(utf8bytes(source_code))
 
+        for path in self.compiler_required_dirs:
+            path = os.path.expanduser(path)
+            assert path == os.path.abspath(path), 'compiler_required_dirs must contain only absolute normalized paths'
```

Comment:
Wait but... they don't? `~/.cache` isn't an absolute path. I don't think it's worth having any restriction here.

Reply:
What happens if the path is not absolute after `expanduser`? What do we do?

Reply:
It's fine, these are not adversarial conditions. We specify the paths.

Reply:
Should I change the error message? I'm very hesitant to allow non-absolute paths.

Reply:
Why? What are you protecting against? This code reads to me like having a check for the sake of having a check, it doesn't provide utility.

Reply:
I've changed the error message.

Reply:
We have sanity checks in our FilesystemAccessRule, right? Why not here?

Reply:
FilesystemAccessRule can be specified by users who may not know what they're doing in `extra_fs`, these can't

Reply:
I've removed the check, but I'm still unsure where the line is for making these checks. Don't we specify the paths for the rules too? I suppose not _always_.